### PR TITLE
docs: add missing Admin API Plugin endpoints

### PR DESCRIPTION
**Pull request by shreemaan-abhishek**
### Description

Fixes # (issue)

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by kayx23** [comment]
I think subsystem could be `http` as well, please test and confirm.

Relevant PR: https://github.com/apache/apisix/pull/9580

File: docs/en/latest/admin-api.md [hunk 1]
```diff
@@ -1312,6 +1312,9 @@ Plugin resource request address: /apisix/admin/plugins/{plugin_name}
 | ------ | ----------------------------------- | ------------ | ---------------------------------------------- |
 | GET    | /apisix/admin/plugins/list          | NULL         | Fetches a list of all Plugins.                 |
 | GET    | /apisix/admin/plugins/{plugin_name} | NULL         | Fetches the specified Plugin by `plugin_name`. |
+| GET         | /apisix/admin/plugins?all=true      | NULL         | Get all properties of all plugins. |
+| GET         | /apisix/admin/plugins?all=true&subsystem=stream| NULL | Get properties of all Stream plugins.|
```
Comment (kayx23) [inline docs/en/latest/admin-api.md:1316]:
```suggestion
| GET         | /apisix/admin/plugins?all=true&subsystem=stream| NULL | Gets properties of all Stream plugins.|
```
...and the one above as well for consistency

File: docs/zh/latest/admin-api.md [hunk 2]
```diff
@@ -1326,6 +1326,7 @@ Plugin 资源请求地址：/apisix/admin/plugins/{plugin_name}
 | GET         | /apisix/admin/plugins/{plugin_name} | 无         | 获取资源。      |
 | GET         | /apisix/admin/plugins?all=true      | 无         | 获取所有插件的所有属性。 |
 | GET         | /apisix/admin/plugins?all=true&subsystem=stream| 无 | 获取所有 Stream 插件的属性。|
```
Comment (kayx23) [inline docs/zh/latest/admin-api.md:1328]:
```suggestion
| GET         | /apisix/admin/plugins?all=true&subsystem=stream| 无 | 获取所有 Stream 插件的属性。|
| GET         | /apisix/admin/plugins?all=true&subsystem=http| 无 | 获取所有 HTTP 插件的属性。|
```

**Comment by kayx23** [comment]
Please also add the doc for "checking plugin in subsystem" (i.e.`...{plugin_name}/subsystem=...`) as seen in TEST 15 in https://github.com/apache/apisix/pull/9580

image

**Comment by kayx23** [comment]
Please see if anything else needs to be updated here @Revolyssup . 

Do we keep the `all=true` here?

> `/apisix/admin/plugins?all=true` API to get all properties of all plugins. This API will be deprecated soon.

**Comment by Revolyssup** [comment]
@kayx23 @shreemaan-abhishek I think we can add the note in the docs that says `?all=true` is not recommended and will be deprecated.  Rest everything looks good to me. And the only remaining thing to add is https://github.com/apache/apisix/pull/9050#issuecomment-1687283884

**Review by Revolyssup — CHANGES_REQUESTED** [review]

File: docs/en/latest/admin-api.md [hunk 3]
```diff
@@ -1316,6 +1316,8 @@ Plugin resource request address: /apisix/admin/plugins/{plugin_name}
 | GET         | /apisix/admin/plugins?all=true&subsystem=stream| NULL | Gets properties of all Stream plugins.|
 | GET    | /apisix/admin/plugins?all=true&subsystem=http | NULL | Gets properties of all HTTP plugins. |
 | PUT    | /apisix/admin/plugins/reload       | NULL         | Reloads the plugin according to the changes made in code |
+| GET    | apisix/admin/plugins/{PLUGIN}?subsystem=stream | NULL | Checks if the `{PLUGIN}` is supported in Stream/L4 mode. |
```
[the remark this answers is not in the document]
Comment (kayx23) [reply docs/en/latest/admin-api.md:1319]:
```
| GET    | /apisix/admin/plugins?all=true&subsystem=http | NULL | Gets properties of all HTTP plugins. |
```

Note the wording here is `Gets properties of`. Please make sure we're consistent (change either one of those).

[the remark this answers is not in the document]
Comment (shreemaan-abhishek) [reply docs/en/latest/admin-api.md:1319]: > I think it should be `Fetches the specified stream Plugin by `plugin_name` ` `Fetches the specified HTTP Plugin by `plugin_name` `

why? 🤔 It didn't make sense to me.

[the remark this answers is not in the document]
Comment (Revolyssup) [reply docs/en/latest/admin-api.md:1319]: @shreemaan-abhishek This endpoint not only checks if the plugin exists in that subsystem but returns the plugin if it does exist in that subsystem.

[the remark this answers is not in the document]
Comment (shreemaan-abhishek) [reply docs/en/latest/admin-api.md:1319]: So is this better?

`Gets the {PLUGIN} if it is supported in XYZ mode`

[the remark this answers is not in the document]
Comment (kayx23) [reply docs/en/latest/admin-api.md:1319]: > ... but returns the plugin if it does exist in that subsystem.

@shreemaan-abhishek you can do a test to see the output . If I recall correctly it returns the plugin **schema**.

**Review by Revolyssup — DISMISSED** [review]

File: docs/zh/latest/admin-api.md [hunk 4]
```diff
@@ -1326,6 +1326,8 @@ Plugin 资源请求地址：/apisix/admin/plugins/{plugin_name}
...
+| GET         | /apisix/admin/plugins?all=true&subsystem=http| 无 | 获取所有 HTTP 插件的属性。|
+| PUT         | /apisix/admin/plugins/reload        | 无         | 根据代码中所做的更改重新加载插件 |
```
Comment (kayx23) [inline docs/zh/latest/admin-api.md:1330]:
```suggestion
| PUT         | /apisix/admin/plugins/reload        | 无         | 根据代码中所做的更改重新加载插件。 |
| GET         | apisix/admin/plugins/{plugin_name}?subsystem=stream         | 无         | 获取指定 Stream 插件的属性。 |
| GET         | apisix/admin/plugins/{plugin_name}?subsystem=http         | 无         | 获取指定 HTTP 插件的属性。 |
```

File: docs/en/latest/admin-api.md [hunk 5]
```diff
@@ -1312,6 +1312,12 @@ Plugin resource request address: /apisix/admin/plugins/{plugin_name}
 | ------ | ----------------------------------- | ------------ | ---------------------------------------------- |
 | GET    | /apisix/admin/plugins/list          | NULL         | Fetches a list of all Plugins.                 |
 | GET    | /apisix/admin/plugins/{plugin_name} | NULL         | Fetches the specified Plugin by `plugin_name`. |
+| GET         | /apisix/admin/plugins?all=true      | NULL         | Get all properties of all plugins. |
+| GET         | /apisix/admin/plugins?all=true&subsystem=stream| NULL | Gets properties of all Stream plugins.|
+| GET    | /apisix/admin/plugins?all=true&subsystem=http | NULL | Gets properties of all HTTP plugins. |
+| PUT    | /apisix/admin/plugins/reload       | NULL         | Reloads the plugin according to the changes made in code |
+| GET    | apisix/admin/plugins/{PLUGIN}?subsystem=stream | NULL | Gets the `{PLUGIN}` schema if it is supported in Stream/L4 subsystem. |
```
Comment (kayx23) [inline docs/en/latest/admin-api.md:1319]:
```suggestion
| GET    | apisix/admin/plugins/{plugin_name}?subsystem=stream | NULL | Gets properties of a specified plugin if it is supported in Stream/L4 subsystem. |
```

File: docs/en/latest/admin-api.md [hunk 6]
```diff
@@ -1312,6 +1312,12 @@ Plugin resource request address: /apisix/admin/plugins/{plugin_name}
...
+| GET    | apisix/admin/plugins/{PLUGIN}?subsystem=http   | NULL | Gets the `{PLUGIN}` schema if it is supported in HTTP/L7 subsystem. |
```
Comment (kayx23) [inline docs/en/latest/admin-api.md:1320]:
```suggestion
| GET    | apisix/admin/plugins/{PLUGIN}?subsystem=http   | NULL | Gets properties of a specified plugin if it is supported in HTTP/L7 subsystem. |
```

File: docs/en/latest/admin-api.md [hunk 7]
```diff
@@ -1312,6 +1312,12 @@ Plugin resource request address: /apisix/admin/plugins/{plugin_name}
 | ------ | ----------------------------------- | ------------ | ---------------------------------------------- |
 | GET    | /apisix/admin/plugins/list          | NULL         | Fetches a list of all Plugins.                 |
 | GET    | /apisix/admin/plugins/{plugin_name} | NULL         | Fetches the specified Plugin by `plugin_name`. |
+| GET         | /apisix/admin/plugins?all=true      | NULL         | Get all properties of all plugins. |
```
Comment (kayx23) [inline docs/en/latest/admin-api.md:1315]: Add deprecate note for `all=true`.

@Revolyssup btw, does `apisix/admin/plugins` do what `/apisix/admin/plugins?all=true` do after the change? Or is the subsystem URL param _required_ for the`/apisix/admin/plugins` endpoint

**Review by kayx23 — CHANGES_REQUESTED** [review]

File: docs/en/latest/admin-api.md (hunk 7 above)
Comment (Revolyssup) [reply docs/en/latest/admin-api.md:1315]: @kayx23 The only way to return all results is with `all=true`. Even with `subsystem`, plugin name has to be provided. The mechanism of getting all plugins at once was deprecated with the usage of all=true.

File: docs/en/latest/admin-api.md [hunk 8]
```diff
@@ -1312,6 +1312,12 @@ Plugin resource request address: /apisix/admin/plugins/{plugin_name}
...
+| GET         | /apisix/admin/plugins?all=true&subsystem=stream| NULL | Gets properties of all Stream plugins.|
+| GET    | /apisix/admin/plugins?all=true&subsystem=http | NULL | Gets properties of all HTTP plugins. |
+| PUT    | /apisix/admin/plugins/reload       | NULL         | Reloads the plugin according to the changes made in code |
+| GET    | apisix/admin/plugins/{plugin_name}?subsystem=stream | NULL | Gets properties of a specified plugin if it is supported in Stream/L4 subsystem. |
+| GET    | apisix/admin/plugins/{PLUGIN}?subsystem=http   | NULL | Gets properties of a specified plugin if it is supported in HTTP/L7 subsystem. |
```
Comment (kayx23) [inline docs/en/latest/admin-api.md:1320]:
```suggestion
| GET    | apisix/admin/plugins/{plugin_name}?subsystem=http   | NULL | Gets properties of a specified plugin if it is supported in HTTP/L7 subsystem. |

:::caution

The interface of getting properties of all plugins via `/apisix/admin/plugins?all=true` will be deprecated soon.

:::
```

File: docs/zh/latest/admin-api.md [hunk 9]
```diff
@@ -1326,6 +1326,10 @@ Plugin 资源请求地址：/apisix/admin/plugins/{plugin_name}
 | GET         | /apisix/admin/plugins/{plugin_name} | 无         | 获取资源。      |
 | GET         | /apisix/admin/plugins?all=true      | 无         | 获取所有插件的所有属性。 |
 | GET         | /apisix/admin/plugins?all=true&subsystem=stream| 无 | 获取所有 Stream 插件的属性。|
+| GET         | /apisix/admin/plugins?all=true&subsystem=http| 无 | 获取所有 HTTP 插件的属性。|
+| PUT         | /apisix/admin/plugins/reload        | 无         | 根据代码中所做的更改重新加载插件。 |
+| GET         | apisix/admin/plugins/{plugin_name}?subsystem=stream         | 无         | 获取指定 Stream 插件的属性。 |
+| GET         | apisix/admin/plugins/{plugin_name}?subsystem=http         | 无         | 获取指定 HTTP 插件的属性。 |
```
Comment (kayx23) [inline docs/zh/latest/admin-api.md:1332]:
```suggestion
| GET         | apisix/admin/plugins/{plugin_name}?subsystem=http         | 无         | 获取指定 HTTP 插件的属性。 |

:::caution

获取所有插件属性的接口 `/apisix/admin/plugins?all=true` 将很快被弃用。

:::
```

**Review by Revolyssup — APPROVED** [review]

**Review by kayx23 — APPROVED** [review]

**Pull request merged**